src/test/java/com/amazon/sqs/javamessaging/AmazonSQSExtendedClientDeprecatedTest.java

```diff
@@ -0,0 +1,231 @@
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.AmazonSQSClient;
+import com.amazonaws.services.sqs.model.MessageAttributeValue;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageResult;
+import com.amazonaws.services.sqs.model.SendMessageBatchRequest;
+import com.amazonaws.services.sqs.model.SendMessageBatchRequestEntry;
+import com.amazonaws.services.sqs.model.SendMessageRequest;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.isA;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests the AmazonSQSExtendedClient deprecated functions.
+ */
+public class AmazonSQSExtendedClientDeprecatedTest {
+    private AmazonSQS extendedSqsWithDefaultConfig;
+    private AmazonSQS mockSqsBackend;
+    private AmazonS3 mockS3;
+    private static final String S3_BUCKET_NAME = "test-bucket-name";
+    private static final String SQS_QUEUE_URL = "test-queue-url";
+
+    private static final int LESS_THAN_SQS_SIZE_LIMIT = 3;
+    private static final int SQS_SIZE_LIMIT = 262144;
+    private static final int MORE_THAN_SQS_SIZE_LIMIT = SQS_SIZE_LIMIT + 1;
+
+    // should be > 1 and << SQS_SIZE_LIMIT
+    private static final int ARBITRATY_SMALLER_THRESSHOLD = 500;
+
+    @Before
+    public void setupClient() {
+        mockS3 = mock(AmazonS3.class);
+        mockSqsBackend = mock(AmazonSQS.class);
+        when(mockS3.putObject(isA(PutObjectRequest.class))).thenReturn(null);
+
+        ExtendedClientConfiguration extendedClientConfiguration = new ExtendedClientConfiguration()
+                .withLargePayloadSupportEnabled(mockS3, S3_BUCKET_NAME);
+
+        extendedSqsWithDefaultConfig = spy(new AmazonSQSExtendedClient(mockSqsBackend, extendedClientConfiguration));
+
+    }
+
+    @Test
+    public void testWhenSendLargeMessageThenPayloadIsStoredInS3() {
+        String messageBody = generateStringWithLength(MORE_THAN_SQS_SIZE_LIMIT);
+
+        SendMessageRequest messageRequest = new SendMessageRequest(SQS_QUEUE_URL, messageBody);
+        extendedSqsWithDefaultConfig.sendMessage(messageRequest);
+
+        verify(mockS3, times(1)).putObject(isA(PutObjectRequest.class));
+    }
+
+    @Test
+    public void testWhenSendSmallMessageThenS3IsNotUsed() {
+        int messageLength = SQS_SIZE_LIMIT;
+        String messageBody = generateStringWithLength(messageLength);
+
+        SendMessageRequest messageRequest = new SendMessageRequest(SQS_QUEUE_URL, messageBody);
+        extendedSqsWithDefaultConfig.sendMessage(messageRequest);
+
+        verify(mockS3, never()).putObject(isA(PutObjectRequest.class));
+    }
+
```

Comment:
This method is repeated in both :
AmazonSQSExtendedClientDeprecatedTest and AmazonSQSExtendedClientTest. 

Are we testing a deprecated method here?